src/joystick_interrupt.cpp

```diff
@@ -56,48 +56,74 @@ class JoystickInterrupt
 
   void cbJoy(const sensor_msgs::Joy::Ptr msg)
   {
-    if (msg->buttons[interrupt_button_])
+    if (!rangeCheck(msg->buttons, interrupt_button_))
     {
-      last_joy_msg_ = ros::Time::now();
+      ROS_ERROR("Out of Range error: buttons size: %lu <= interrupt_button_: %d",
+                msg->buttons.size(), interrupt_button_);
+      last_joy_msg_ = ros::Time(0);
+      return;
+    }
+    if (!msg->buttons[interrupt_button_])
+    {
+      last_joy_msg_ = ros::Time(0);
+      return;
+    }
 
-      float lin, ang;
+    last_joy_msg_ = ros::Time::now();
+
+    float lin(0.0f);
+    float ang(0.0f);
+    if (rangeCheck<float>(msg->axes, linear_axis_))
       lin = msg->axes[linear_axis_];
+    else
+      ROS_ERROR("Out of range error: axes size: %lu <= linear_axis: %d", msg->axes.size(), linear_axis_);
+    if (rangeCheck<float>(msg->axes, angular_axis_))
       ang = msg->axes[angular_axis_];
-      if (linear_axis2_ >= 0)
+    else
+      ROS_ERROR("Out of range error: axes size: %lu <= angular_axis: %d", msg->axes.size(), angular_axis_);
+
+    if (isEnabled(linear_axis2_))
+    {
+      if (rangeCheck<float>(msg->axes, linear_axis2_))
       {
         if (fabs(msg->axes[linear_axis2_]) > fabs(lin))
-        {
           lin = msg->axes[linear_axis2_];
-        }
       }
-      if (angular_axis2_ >= 0)
+      else
+        ROS_ERROR("Out of range error: axes size: %lu <= linear_axis2: %d", msg->axes.size(), linear_axis2_);
+    }
+    if (isEnabled(angular_axis2_))
+    {
+      if (rangeCheck<float>(msg->axes, angular_axis2_))
       {
-        if (fabs(msg->axes[angular_axis2_]) > fabs(lin))
-        {
+        if (fabs(msg->axes[angular_axis2_]) > fabs(ang))
```

Comment:
Fixed from fabs(lin) to fabs(ang)
https://github.com/at-wat/joystick_interrupt/blob/33ffaa8163048a411ae2ad3742b56c58ad5ee468/src/joystick_interrupt.cpp#L75